v0/destinations/personalize/transform.js

```diff
@@ -42,27 +49,62 @@ async function process(event) {
   for (let j = 0; j < noOfFields; j += 1) {
     mappedField.push(event.destination.Config.customMappings[j].to);
   }
-  if (event.message.event === event.destination.Config.eventName) {
+  if (message.event === event.destination.Config.eventName) {
     const property = {};
     for (let k = 0; k < noOfFields; k += 1) {
       if (
         keyField[k].toUpperCase() === "USER_ID" ||
         keyField[k].toUpperCase() === "EVENT_TYPE" ||
         keyField[k].toUpperCase() === "TIMESTAMP"
       ) {
-        keyField[k] = keyField[k].replace(/_([a-z])/g, function(g) {
+        keyField[k] = keyField[k].replace(/_([a-z])/g, function (g) {
           return g[1].toUpperCase();
         });
       } else {
         const mappedFields = mappedField[k];
-
-        if (typeof event.message[mappedFields] !== "undefined") {
-          keyField[k] = keyField[k].replace(/_([a-z])/g, function(g) {
-            return g[1].toUpperCase();
-          });
-          property[keyField[k]] = event.message[mappedFields];
-        } else {
-          throw new Error(`Mapped Field ${mappedFields} not found`);
+        switch (messageType) {
+          case EventType.TRACK:
+          case EventType.PAGE:
+          case EventType.SCREEN:
+            keyField[k] = keyField[k].replace(/_([a-z])/g, function(g) {
+              return g[1].toUpperCase();
+            });
+            if (message.properties && message.properties[mappedFields]) {
```

Comment:
we can save the value of message.properties[mappedFields] if message.properties is present so that the code does not need to get the value again in the next line.